src/main/webapp/app/overview/course-overview.component.ts

```diff
@@ -194,6 +204,94 @@ export class CourseOverviewComponent implements OnInit, OnDestroy, AfterViewInit
         await this.initAfterCourseLoad();
         this.sidebarItems = this.getSidebarItems();
         this.courseActionItems = this.getCourseActionItems();
+        this.updateVisibility(window.innerHeight);
+        this.updateMenuOffset();
+    }
+
+    /** Listen window resizement event by height */
+    @HostListener('window: resize', ['$event'])
+    onResize() {
+        this.dropdownOpen = false;
+        this.dropdownClickNumber = 0;
+        this.updateVisibility(window.innerHeight);
+        this.updateMenuOffset();
+    }
+
+    /** Listen click event whether on outside of the menu or one of the items in the menu to close the dropdown menu */
+    @HostListener('document: click', ['$event'])
+    onClickCloseDropdownMenu() {
+        if (this.dropdownOpen) {
+            this.dropdownClickNumber += 1;
+            if (this.dropdownClickNumber === 2) {
+                this.dropdownOpen = false;
+                this.dropdownClickNumber = 0;
+            }
+        }
+    }
+
+    /** Update sidebar item's hidden property based on the window height to display three-dots */
+    updateVisibility(height: number) {
+        const thresholds: number[] = [];
+        const itemHeight: number = 35;
+        let threshold = this.calculateThreshold();
+        this.anyItemHidden = false;
+        this.hiddenItems = [];
+
+        for (let i = 0; i < this.sidebarItems.length - 1; i++) {
+            thresholds.unshift(threshold);
+            threshold -= itemHeight;
+        }
+        thresholds.unshift(0);
+
+        this.sidebarItems.forEach((item, index) => {
+            item.hidden = height <= thresholds[index];
+            if (item.hidden) {
+                this.anyItemHidden = true;
+                this.hiddenItems.push(item);
+            }
+        });
+    }
+
+    /** Calculate dropdown-menu position based on the number of entries in the sidebar */
+    updateMenuOffset() {
+        const leftSidebarItems: number = this.sidebarItems.length - this.hiddenItems.length;
+        const minHeightForSidebar: number = 330; // Minimum height for the sidebar to dynamically position dropdown menu
+        const offsetForEachElement: number = 40; // Offset to be added to the dropdown menu for each element left in the sidebar
+        const maxSidebarElements: number = 9;
```

Comment:
For constants usually uppercase is used, so you directly know that this will not be modified
```suggestion
        const MIN_SIDEBAR_HEIGHT: number = 330; // Minimum height for the sidebar to dynamically position dropdown menu
        const SIDEBAR_ELEMENT_OFFSET: number = 40; // Offset to be added to the dropdown menu for each element left in the sidebar
        const MAX_SIDEBAR_ELEMENTS: number = 9;
```

I would also suggest to define it at the top, which would make the methods easier to read and allow to focus on the actual logic of the method.

I do also think that with the descriptive Variable names the comments are not needed.